diff --git a/interfaces/rust/rustc_demangle/BUILD.gn b/interfaces/rust/rustc_demangle/BUILD.gn
index 8cecd6b..5b06ab3 100644
--- a/interfaces/rust/rustc_demangle/BUILD.gn
+++ b/interfaces/rust/rustc_demangle/BUILD.gn
@@ -13,36 +13,15 @@
 
 import("//base/hiviewdfx/faultloggerd/faultloggerd.gni")
 
-ohos_rust_static_ffi("rustc_demangle_static") {
+ohos_rust_shared_ffi("rustc_demangle") {
   include_dirs = [ "." ]
   sources = [ "lib.rs" ]
-  if (current_cpu == "arm64") {
-    externs = [
-      {
-        crate_name = "rustc_demangle"
-        path = "//prebuilts/rustc/linux-x86_64/current/lib/rustlib/aarch64-linux-ohos/lib/librustc_demangle-7093163c24b92e64.rlib"
-      },
-    ]
-  } else if (current_cpu == "arm") {
-    externs = [
-      {
-        crate_name = "rustc_demangle"
-        path = "//prebuilts/rustc/linux-x86_64/current/lib/rustlib/arm-linux-ohos/lib/librustc_demangle-aff4ef91724960ce.rlib"
-      },
-    ]
-  } else if (current_cpu == "x86_64" || current_cpu == "x64") {
-    externs = [
-      {
-        crate_name = "rustc_demangle"
-        path = "//prebuilts/rustc/linux-x86_64/current/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-46c8c3eb5be77633.rlib"
-      },
-    ]
-  } else {
-    externs = []
-  }
-
   crate_name = "rustc_demangle"
-  crate_type = "staticlib"
+  crate_type = "cdylib"
+  install_images = [
+    "system",
+    "updater"
+  ]
   part_name = "faultloggerd"
   subsystem_name = "hiviewdfx"
 }
diff --git a/test/unittest/rustc_demangle/BUILD.gn b/test/unittest/rustc_demangle/BUILD.gn
index 480b63d..40c8925 100644
--- a/test/unittest/rustc_demangle/BUILD.gn
+++ b/test/unittest/rustc_demangle/BUILD.gn
@@ -42,7 +42,7 @@ if (defined(ohos_lite)) {
     sources = [ "rustc_demangle_test.cpp" ]
 
     deps = [
-      "$faultloggerd_path/interfaces/rust/rustc_demangle:rustc_demangle_static",
+      "$faultloggerd_path/interfaces/rust/rustc_demangle:rustc_demangle",
       "//third_party/googletest:gtest_main",
     ]
 
diff --git a/tools/process_dump/BUILD.gn b/tools/process_dump/BUILD.gn
index e67d8e4..ecfa9e3 100644
--- a/tools/process_dump/BUILD.gn
+++ b/tools/process_dump/BUILD.gn
@@ -158,7 +158,7 @@ if (defined(ohos_lite)) {
       "$faultloggerd_path/common/dfxlog:static_common_log",
       "$faultloggerd_path/common/dfxutil:dfx_util",
       "$faultloggerd_path/interfaces/innerkits/faultloggerd_client:libfaultloggerd",
-      "$faultloggerd_path/interfaces/rust/rustc_demangle:rustc_demangle_static",
+      "$faultloggerd_path/interfaces/rust/rustc_demangle:rustc_demangle",
       "//third_party/libunwind:libunwind",
     ]
 
@@ -198,7 +198,7 @@ if (defined(ohos_lite)) {
       "$faultloggerd_path/common/dfxutil:dfx_util",
       "$faultloggerd_path/interfaces/innerkits/faultloggerd_client:libfaultloggerd",
       "$faultloggerd_path/interfaces/innerkits/signal_handler:dfx_local_handler",
-      "$faultloggerd_path/interfaces/rust/rustc_demangle:rustc_demangle_static",
+      "$faultloggerd_path/interfaces/rust/rustc_demangle:rustc_demangle",
       "//third_party/libunwind:libunwind",
     ]
 
